src/main/java/org/commcare/formplayer/session/PersistentMenuHelper.kt

```diff
@@ -36,10 +36,9 @@ class PersistentMenuHelper {
      fun advanceCurrentMenuWithInput(screen: Screen, input: String) {
         if (screen is MenuScreen) {
             val index = input.toInt()
-            if (currentMenu != null && index < currentMenu!!.commands.size) {
-                currentMenu =  currentMenu!!.commands[index]
-            } else if (index < persistentMenu!!.size) {
-                currentMenu =  persistentMenu!![index]
+            val parentMenu = if (currentMenu == null) persistentMenu else currentMenu!!.commands
```

Comment:
why is the !! needed if we already know that currentMenu is not null?

Reply:
Compiler can't really confirm non nullability in a multi-threaded environment for a variable, therefore it always needs us to supply `!!` cast for non nullability.